src/main/core/worker.h

```diff
@@ -52,6 +52,22 @@ void workerpool_joinAll(WorkerPool* pool);
 void workerpool_free(WorkerPool* pool);
 pthread_t workerpool_getThread(WorkerPool* pool, int threadId);
 
+// Compute the global min event time across all workers. We dynamically compute
+// the minimum time that we'll need for the next event round as the minimum of
+// i.) all events pushed by all workers during this round, and
+// ii.) the next queued event for all worker at the point when they stop
+// executing events.
+//
+// This func is not thread safe, so only call from the scheduler thread when the
+// workers are idle.
+SimulationTime workerpool_getGlobalNextEventTime(WorkerPool* workerPool);
+
+// The worker either pushed an event or finished executing its events and is
+// reporting the min time of events in their event queue.
```

Comment:
It's currently the caller's responsibility to only call with `simtime < currentRoundEndTime`. Should note that here if we keep it that way, but I think it'd be better to move that check inside this function.

Reply:
Moved check to `worker_setMinEventTimeNextRound`.